testimony/__init__.py

```diff
@@ -26,34 +27,165 @@
 }
 
 
-class Result(object):
-    """Class that represents a path result"""
+def indent(text, prefix, predicate=None):
+    """Adds 'prefix' to the beginning of selected lines in 'text'.
 
-    def __init__(self, bugs=0, bugs_list=None, invalid_doc=0,
-                 no_docstring=0, no_minimal_docstring=0, manual_count=0,
-                 tc_count=0):
-        """bugs: number of bugs found
-        bugs_list: list of bugs found
-        invalid_doc: number of testcases with invalid docstrings found
-        no_docstring: number of testcases with no docstring
-        no_minimal_docstring: number of testcases that don't have at least
-            feature, test and assert tags
-        manual_count: number of testcases that represents manual testing
-        tc_count: total number of testcases found
+    If 'predicate' is provided, 'prefix' will only be added to the lines
+    where 'predicate(line)' is True. If 'predicate' is not provided,
+    it will default to adding 'prefix' to all non-empty lines that do not
+    consist solely of whitespace characters.
 
+    PS.: From Python 3.3+ textwrap.indent.
+    """
+    if predicate is None:
+        def predicate(line):
+            return line.strip()
+
+    def prefixed_lines():
```

Comment:
I just used python 3 implementation of that, the idea here is to define this just when it is python 2.